flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/PrioritizedOperatorSubtaskStateTest.java

```diff
@@ -51,6 +55,108 @@ class PrioritizedOperatorSubtaskStateTest {
 
     private static final Random RANDOM = new Random(0x42);
 
+    @Test
+    void testTryCreateMixedLocalAndRemoteAlternative() {
+        testTryCreateMixedLocalAndRemoteAlternative(
+                StateHandleDummyUtil::createKeyedStateHandleFromSeed,
+                KeyedStateHandle::getKeyGroupRange);
+    }
+
+    <SH extends StateObject, ID> void testTryCreateMixedLocalAndRemoteAlternative(
+            IntFunction<SH> stateHandleFactory, Function<SH, ID> idExtractor) {
+        List<SH> jmState =
+                Arrays.asList(
+                        stateHandleFactory.apply(0),
+                        stateHandleFactory.apply(1),
+                        stateHandleFactory.apply(2),
+                        stateHandleFactory.apply(3));
+
+        List<SH> alternativeA =
+                Arrays.asList(stateHandleFactory.apply(0), stateHandleFactory.apply(3));
+
+        List<SH> alternativeB =
+                Arrays.asList(
+                        stateHandleFactory.apply(1),
+                        stateHandleFactory.apply(3),
+                        stateHandleFactory.apply(5));
+
+        List<StateObjectCollection<SH>> alternatives =
+                Arrays.asList(
+                        new StateObjectCollection<>(alternativeA),
+                        new StateObjectCollection<>(Collections.emptyList()),
+                        new StateObjectCollection<>(alternativeB));
+
+        StateObjectCollection<SH> result =
+                PrioritizedOperatorSubtaskState.Builder.tryComputeMixedLocalAndRemoteAlternative(
+                                new StateObjectCollection<>(jmState), alternatives, idExtractor)
+                        .get();
+
+        Assertions.assertEquals(4, result.size());
+        List<SH> expected = new ArrayList<>(alternativeA);
+        expected.add(alternativeB.get(0));
+        expected.add(jmState.get(2));
+        Assertions.assertTrue(result.containsAll(expected));
```

Comment:
I find this test more difficult to follow than it could be.
How about:
1. Use `org.assertj.core.api.Assertions.assertThat(result).hasSameElementsAs(...)`
2. Use constants for handles:
```
SH remote0 = stateHandleFactory.apply(0);
SH remote1 = stateHandleFactory.apply(1);
...
List<SH> jmState = Arrays.asList( remote0, remote1 , remote2 , remote3 );

SH local0 = stateHandleFactory.apply(0);
SH local3 = stateHandleFactory.apply(3);

...
List<SH> alternativeA = Arrays.asList(...);
List<SH> alternativeB = Arrays.asList(...);
...
assertThat(result).hasSameElementsAs(Arrays.asList(local0, local3, local1, remote0));